### DISCUSSION TITLE
Using github actions to publish to github package repository

### DISCUSSION THREAD

**Discussion by jtratner** [community/community#25916] [Actions]
<p>I’m trying to use this workflow to package a release to github package repository.  But currently when I run it the tests pass but I get a 401. Any ideas?</p>
<pre><code>name: Release workflow

on:
  push:
    # Sequence of patterns matched against refs/tags
    tags:
      - 'v*' # Push events to matching v*, i.e. v1.0, v20.15.10
jobs:
  build:
    name: Create Release
    runs-on: ubuntu-latest
    steps:
      - name: Checkout code
        uses: actions/checkout@v2
      # TODO: Can I just download the artifact from the previous commit run??
      - name: Set up JDK 1.8
        uses: actions/setup-java@v1
        with:
          java-version: 1.8
      - uses: actions/cache@v1
        with:
          path: ~/.m2/repository
          key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
          restore-keys: |
            ${{ runner.os }}-maven-
      - name: Build with Maven
        run: mvn -B package --file pom.xml
      - name: push to github packages with maven
        continue-on-error: true
        run: mvn -B deploy --file pom.xml
      - name: clean up target folder
        run: rm -rf ./target/lib ./target/test-classes
      - name: Upload artifact
        uses: actions/upload-artifact@v1.0.0
        # TODO: make this only run if the build step at least ran
        if: always()
        with:
          # Artifact name
          name: auto-pipeline-jar
          # Directory containing files to upload
          path: ./target
      - name: Create zip
        run: cd target; zip -r ../target.zip .
      - name: Create Release
        id: create_release
        uses: actions/create-release@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # This token is provided by Actions, you do not need to create your own token
        with:
          tag_name: ${{ github.ref }}
          release_name: Release ${{ github.ref }}
          body: New release for ${{ github.ref }}
          draft: false
          prerelease: false
      - name: Upload Release Asset
        id: upload-release-asset
        uses: actions/upload-release-asset@v1
        env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        with:
          upload_url: ${{ steps.create_release.outputs.upload_url }} # This pulls from the CREATE RELEASE step above, referencing it's ID to get its outputs object, which include a `upload_url`. See this blog post for more info: https://jasonet.co/posts/new-features-of-github-actions/#passing-data-to-future-steps
          asset_path: ./target.zip
          asset_name: target.zip
          asset_content_type: application/zip
      # TODO: Upload the jars individually, but for now this is good enough
</code></pre>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Did you get 401 message in step " <strong>push to github packages with maven</strong>" ? </p>
<p>When you try to deploy maven package in your workflow, you could use a specific settings.xml including your personal access token . You could store the PAT in secrets and use it as an environment variable value when run <em>mvn deploy</em> command. </p>
<p>Please see my example here: </p>
<pre><code>name: deploy maven package
on: push
jobs:
  job1:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2
    - run: mvn deploy --settings settings.xml
      env:
        PackagePAT: ${{secrets.PackagePAT}}
</code></pre>
<p>In my repo, I have a settings.xml in the root . In servers section, I set ${env.PackagePAT} as the password. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/3/3cb2fd91c726af483287d074c27e4b8f26c781c9.png" data-download-href="/uploads/short-url/8EY8hCAPzymQzGhfSZKHIkoNiDn.png?dl=1" title="server password.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181094098-50f12a8b-e5a3-411e-8b3e-6ecffdb8bd8f.png" alt="server password.png" data-base62-sha1="8EY8hCAPzymQzGhfSZKHIkoNiDn" width="474" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/3/3cb2fd91c726af483287d074c27e4b8f26c781c9_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">server password.png</span><span class="informations">489×515 21.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>

**Comment by Yanjingzhu** [comment]
<p><a class="mention" href="/jtratner">@jtratner</a> How are things going now? Please check my previous reply, could specific settings.xml and pass PAT to it work for you ?      </p>

**Comment by bissim** [comment]
<p>Hi <a class="mention" href="/yanjingzhu">@yanjingzhu</a>,<br>
Your <code>settings.xml</code> file worked for me indeed.</p>
